include/exec/helper-head.h

```diff
@@ -48,6 +49,8 @@
 #define dh_ctype_cptr const void *
 #define dh_ctype_void void
 #define dh_ctype_memop MemOp
+/* Can't use TCGMemOpIdx here due to include ordering. */
+#define dh_ctype_memop_idx uint32_t /* TCGMemOpIdx */
```

Comment:
Did you actually verify that? Because this is a #define, TCGMemOpIdx only needs to be defined when this macro is used, not here.

Reply:
```[2/97] Compiling C object libqemu-riscv64cheri-softmmu.fa.p/accel_tcg_tcg-runtime-gvec.c.o
In file included from ../../qemu/accel/tcg/tcg-runtime-gvec.c:23:
In file included from /Users/alex/cheri/qemu/include/exec/helper-proto.h:40:
/Users/alex/cheri/qemu/accel/tcg/tcg-runtime.h:154:1: warning: type specifier missing, defaults to 'int' [-Wimplicit-int]
DEF_HELPER_3(cheri_invalidate_tags, void, env, cap_checked_ptr, memop_idx)
^
/Users/alex/cheri/qemu/include/exec/helper-head.h:163:5: note: expanded from macro 'DEF_HELPER_3'
    DEF_HELPER_FLAGS_3(name, 0, ret, t1, t2, t3)
    ^
/Users/alex/cheri/qemu/include/exec/helper-proto.h:19:57: note: expanded from macro 'DEF_HELPER_FLAGS_3'
dh_ctype(ret) HELPER(name) (dh_ctype(t1), dh_ctype(t2), dh_ctype(t3));
                                                        ^
/Users/alex/cheri/qemu/include/exec/helper-head.h:55:21: note: expanded from macro 'dh_ctype'
#define dh_ctype(t) dh_ctype_##t
                    ^
<scratch space>:217:1: note: expanded from here
dh_ctype_memop_idx
^
/Users/alex/cheri/qemu/include/exec/helper-head.h:53:28: note: expanded from macro 'dh_ctype_memop_idx'
#define dh_ctype_memop_idx TCGMemOpIdx
                           ^
```

Reply:
I guess we could move those types to exec/memop.h but I'd prefer if we keep upstream diffs small.

Reply:
Yeah ok just tried it locally and there are a lot of those cases